omod/src/main/java/org/openmrs/module/spa/web/SpaWebComponentRegistrar.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.spa.web;
+
+import javax.servlet.DispatcherType;
+import javax.servlet.FilterRegistration.Dynamic;
+import javax.servlet.ServletContext;
+import javax.servlet.ServletRegistration;
+import java.util.EnumSet;
+
+import org.directwebremoting.servlet.EfficientShutdownServletContextAttributeListener;
+import org.openmrs.module.spa.filter.SpaFilter;
+import org.openmrs.module.spa.servlet.SpaServlet;
+import org.openmrs.web.SessionListener;
+import org.springframework.stereotype.Component;
+import org.springframework.web.context.ServletContextAware;
+
+@Component
+public class SpaWebComponentRegistrar implements ServletContextAware {
+
+    @Override
+    public void setServletContext(ServletContext servletContext) {
+
+        try {
+            ServletRegistration openmrsServletReg = servletContext.getServletRegistration("openmrs");
```

Comment:
What is this mapping used for?

Reply:
I thought we need to ensure that any mapping is supposed to be on top of the OpenMRS servlet.  just to eliminate circumstances like `https:localhost:port/ws/frontend/config.json`

Reply:
What happens when you remove line 32?

Reply:
If `servletContext.getServletRegistration("openmrs");` isn't necessary @dkayiwa  how can i add mappings to the Openmrs Servlet then

Reply:
What happens when you remove line 32?

Reply:
works fine.